lib/jackal/utils/spec/helpers.rb

```diff
@@ -67,3 +67,21 @@ def run_setup(config)
   source_wait(:setup)
   runner
 end
+
+# Store callback execution flag in payload to test callback validity
+
+# @klass callback class to inject execution tracking
+def track_execution(klass)
+  alias_name = :execute_orig
+  # Ensure this is called only once within test suite
+  return if klass.method_defined?(alias_name)
+
+  klass.send(:alias_method, alias_name, :execute)
+  klass.send(:define_method, :execute) do |message|
+    message.args['message']['executed'] = true
+    execute_orig(message)
+  end
+
+  # add convenience method for checking callback execution
+  Bogo::Smash.send(:define_method, :executed?) { self.get(:executed) == true }
```

Comment:
This method feels too ambiguous to monkey patch into the `Smash` class. Would rather this be just a standalone method that can be called.